### PR TITLE
firewall: accessibility fixes for NAT page

### PR DESCRIPTION
**Pull request by jfayre**
Add aria-label to buttons and links in table.
Added role="cell" in last column of table to prevent screen readers from reading action buttons when navigating to other table rows.

### CODE REVIEW

File: src/www/firewall_nat.php [hunk 1]
```diff
@@ -339,21 +339,21 @@ function delete_id($id, &$array)
                       <th><?=gettext("IP");?></th>
                       <th><?=gettext("Ports");?></th>
                       <th><?=gettext("Description");?></th>
-                      <th class="text-nowrap">
-                        <a href="firewall_nat_edit.php" class="btn btn-primary btn-xs" data-toggle="tooltip" title="<?= html_safe(gettext('Add')) ?>">
+                      <th role="cell" class="text-nowrap">
+                        <a href="firewall_nat_edit.php" class="btn btn-primary btn-xs" data-toggle="tooltip" title="<?= html_safe(gettext('Add')) ?>" aria-label="<?= html_safe(gettext('Add')) ?>"">
```
Comment (AdSchellevis) [inline src/www/firewall_nat.php:343]:
```suggestion
                        <a href="firewall_nat_edit.php" class="btn btn-primary btn-xs" data-toggle="tooltip" title="<?= html_safe(gettext('Add')) ?>" aria-label="<?= html_safe(gettext('Add')) ?>">
```

**Review by AdSchellevis — CHANGES_REQUESTED** [review]

**Comment by AdSchellevis** [comment]
@jfayre 
While reading this PR, I'm wondering if we are trying to apply the right fix here.

Looking at `bootstrap.js` on our end, on rendering the title is moved to `data-original-title`:

https://github.com/opnsense/core/blob/93304298abf4dc48a47766b2fa30729489ec6db8/src/opnsense/www/js/bootstrap.js#L1784-L1789

Newer versions seem to add `aria-label` here too:

https://github.com/twbs/bootstrap/blob/7c392498fabed7b4327e7c798c6af152b5071d29/dist/js/bootstrap.js#L3554-L3564

Maybe we better backport this change on our end so all tooltips are equal and we don't need to copy titles. 

What do you think?

**Comment by fichtner** [comment]
I would prefer the deduplication too. The risk for regressions is also a factor here changing all these lines. Things happen.

**Comment by jfayre** [comment]
I completely agree if you can do that. There are a lot of these types of issues and it would be great if we could fix them all in one go.Sent from my iPhoneOn Jul 8, 2024, at 5:27 AM, Ad Schellevis ***@***.***> wrote:﻿
@jfayre
While reading this PR, I'm wondering if we are trying to apply the right fix here.
Looking at bootstrap.js on our end, on rendering the title is moved to data-original-title:
https://github.com/opnsense/core/blob/93304298abf4dc48a47766b2fa30729489ec6db8/src/opnsense/www/js/bootstrap.js#L1784-L1789
Newer versions seem to add aria-label here too:
https://github.com/twbs/bootstrap/blob/7c392498fabed7b4327e7c798c6af152b5071d29/dist/js/bootstrap.js#L3554-L3564
Maybe we better backport this change on our end so all tooltips are equal and we don't need to copy titles.
What do you think?

**Comment by AdSchellevis** [comment]
I'll take a look

**Comment by AdSchellevis** [comment]
https://github.com/opnsense/core/commit/b0cab3bdca1180826ffe230ccca7e56280529759 should do the trick

**Pull request closed without merging**